src/core/qgsexpressioncontext.cpp

```diff
@@ -769,3 +769,45 @@ QgsFeedback *QgsExpressionContext::feedback() const
 {
   return mFeedback;
 }
+
+QString QgsExpressionContext::uniqueHash( bool &ok, const QSet<QString> &variables ) const
+{
+  QString hash;
+  ok = true;
+  const QString delimiter( QStringLiteral( "||~~||" ) );
+
+  if ( hasFeature() )
+  {
+    hash.append( QString::number( feature().id() ) + delimiter + QString::number( qHash( feature() ) ) + delimiter );
```

Comment:
`id` is already included by qhash, why repeat it?

Reply:
See comment below for the explanation